### DISCUSSION TITLE
Template engine for elysia?

### DISCUSSION THREAD

**Discussion by SercanBayram** [elysiajs/elysia#571] [Q&A]
how can i use template engine like express -handler for elysia? is there a module or method for this?

**Comment by binyamin** [comment]
Are you talking about engines like Nunjucks and Liquid? Express is not a template engine.

**Comment by kravetsone** [comment]
> how can i use template engine like express -handler for elysia? is there a module or method for this?

use the elysiajs/html plugin

**Comment by is-jonreeves** [comment]
I agree, `@elysiajs/html` will allow you to work with [template literals or JSX](https://elysiajs.com/plugins/html), so should cover your needs:
```ts
// /layouts/Page.tsx
import { Html } from '@elysiajs/html';

export const Page = ({ title, children }: { title: string, children: Html.Children }) => (
  <html>
    <head>
      <title>{title}</title>
    </head>
    <body>
      <div id="root">{children}</div>
    </body>
  </html>
);


// /views/Home.tsx
export const Home = ({ fruit }: { fruit: string[] }) => {
  return (
    <>
      <h1>Welcome Home</h1>
      <p>Here are some fruits:</p>
      <ul>
        {fruit.map(f => <li>{f}</li>)}
      </ul>
    </>
  );
};


// /index.tsx
import { Elysia } from 'elysia';
import { html } from '@elysiajs/html';

import { Page } from './layouts/Page.tsx';
import { Home } from './views/Home.tsx';

// Application
const app = new Elysia()
  .use(html())
  .get('/', () => {
    return (
      <Page title="Home">
        <Home fruit={['🍎', '🍊', '🍇', '🍉', '🍌']} />
      </Page>
    );
  })
  .listen(3001);

// Debug
console.log(`🦊 Elysia is running at ${app.server?.hostname}:${app.server?.port}`);
```

**Reply by nachat-ayoub** [reply]
how about if i wanted to use something like Ejs or HBS with props passing like Express offer?

**Reply by kravetsone** [reply]
> how about if i wanted to use something like Ejs or HBS with props passing like Express offer?

Why u want it?
Just build a simple plugin for it

**Reply by nachat-ayoub** [reply]
Thanks, i was needing a simple server + connected client and i did build a method to use HBS with Elysia
